File: SDKs/googleapis.com/monitoring/v3/python/src/sdk/models/shared/performancethreshold.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import basicsli as shared_basicsli
from ..shared import requestbasedsli as shared_requestbasedsli


@dataclass_json
@dataclasses.dataclass
class PerformanceThreshold:
    r"""PerformanceThreshold
    A PerformanceThreshold is used when each window is good when that window has a sufficiently high performance.
    """
    
    basic_sli_performance: Optional[shared_basicsli.BasicSli] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('basicSliPerformance') }})
    performance: Optional[shared_requestbasedsli.RequestBasedSli] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('performance') }})
    threshold: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('threshold') }})
    

File: SDKs/googleapis.com/monitoring/v3/java/lib/src/main/java/openapisdk/models/shared/PerformanceThreshold.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * PerformanceThreshold
 * A PerformanceThreshold is used when each window is good when that window has a sufficiently high performance.
**/
public class PerformanceThreshold {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("basicSliPerformance")
    public BasicSli basicSliPerformance;
    public PerformanceThreshold withBasicSliPerformance(BasicSli basicSliPerformance) {
        this.basicSliPerformance = basicSliPerformance;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("performance")
    public RequestBasedSli performance;
    public PerformanceThreshold withPerformance(RequestBasedSli performance) {
        this.performance = performance;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("threshold")
    public Double threshold;
    public PerformanceThreshold withThreshold(Double threshold) {
        this.threshold = threshold;
        return this;
    }
}